File: 은지/LV1/숫자 문자열과 영단어.py
def solution(s):
    answer = ""
    num = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    n = ""
    for v in s:
        if v.isdigit():
            answer += v
        else:
            n += v

        if n in num:
            for i in range(10):
                if n == num[i]:
                    answer += str(i)
            n = ""

    answer = int(answer)
    return answer

s = "zeroone"
print(solution(s))

File: 은지/LV1/숫자 문자열과 영단어.js
function solution(s) {
    var answer = 0;
    const num = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    var n = ""
    for (let v of s) {
      if (isNaN(v)) {
        n += v
      } else {
        answer += v
      }

      for (let i=0; i<10; i++) {
        if (n === num[i]) {
          answer += String(i)
          n = ""
          break
        }
      }
    }
    answer = Number(answer)
    return answer;
}

s = "zeroone"
console.log(solution(s))